apps/web/pages/booking/direct/[...link].tsx

```diff
@@ -52,50 +54,57 @@ const requestSchema = z.object({
           message: pageErrors.signature_mismatch,
           code: "custom",
         });
-        console.log(signedData, data, data[3], "==", sig);
       }
     }),
   reason: z.string().optional(),
 });
 
-function bookingContent(status: BookingStatus | undefined | null) {
+function bookingContent(status: BookingStatus | undefined | null, t: TFunction) {
   switch (status) {
     case BookingStatus.PENDING:
       // Trying to reject booking without reason
       return {
         iconColor: "gray",
         Icon: Icon.FiCalendar,
-        titleKey: "event_awaiting_approval",
-        subtitleKey: "someone_requested_an_event",
+        title: t("event_awaiting_approval"),
+        subtitle: t("someone_requested_an_event"),
       };
     case BookingStatus.ACCEPTED:
       // Booking was acepted successfully
       return {
         iconColor: "green",
         Icon: Icon.FiCheck,
-        titleKey: "booking_confirmed",
-        subtitleKey: "emailed_you_and_any_other_attendees",
+        title: t("booking_confirmed"),
+        subtitle: t("emailed_you_and_any_other_attendees"),
       };
     case BookingStatus.REJECTED:
       // Booking was rejected successfully
       return {
         iconColor: "red",
         Icon: Icon.FiX,
-        titleKey: "booking_rejection_success",
-        subtitleKey: "emailed_you_and_any_other_attendees",
+        title: t("booking_rejection_success"),
+        subtitle: t("emailed_you_and_any_other_attendees"),
       };
     default:
       // Booking was already accepted or rejected
       return {
         iconColor: "yellow",
         Icon: Icon.FiAlertTriangle,
-        titleKey: "booking_already_accepted_rejected",
+        title: t("booking_already_accepted_rejected"),
       };
   }
 }
 
-export default function Directlink({ booking, reason, status }: inferSSRProps<typeof getServerSideProps>) {
-  const { t } = useLocale();
+export default function Directlink({
+  booking,
+  reason,
+  status,
+  translations,
+}: inferSSRProps<typeof getServerSideProps>) {
+  let { t } = useLocale();
+  if (!t) {
+    t = (key: string) => translations[key];
```

Comment:
Why is this needed? 😕

Reply:
That's my take on server side translations, maybe I did it wrong? 🤔

Reply:
I'm not 100% onboard with this approach either. What specific problem this is solving if I may ask?

Reply:
As pointed out in a thread, direct link lacks a social preview although I don't think it's really needed, who would share an accept/reject link? 😅